packages/builder/src/package.ts

```diff
@@ -112,6 +112,48 @@ export async function copyPackage({ packageRef, tags, variant, fromStorage, toSt
   }
 }
 
+/**
+ * Recursively reads the given package from IPFS and makes sure to pin it on the publishStorage.
+ */
+export async function publishIpfs({
+  packageRef,
+  chainId,
+  preset,
+  readStorage,
+  publishStorage,
+}: {
+  packageRef: string;
+  chainId: number;
+  preset: string;
+  readStorage: CannonStorage;
+  publishStorage: CannonStorage;
+}) {
+  const deployData = await readStorage.readDeploy(packageRef, preset, chainId);
+
+  if (!deployData) {
+    throw new Error('ipfs could not find deployment artifact. please double check your settings, and rebuild your package.');
+  }
+
+  const metaUrl = await readStorage.registry.getMetaUrl(packageRef, `${chainId}-${preset}`);
+
+  const copyIpfs = async (deployInfo: DeploymentInfo) => {
+    const newMiscUrl = await readStorage.putBlob(await readStorage.readBlob(deployInfo.miscUrl));
```

Comment:
```suggestion
    const newMiscUrl = await publishStorage.putBlob(await readStorage.readBlob(deployInfo.miscUrl));
```
@mjlescano I am not well-versed on this, am I right?

Reply:
Besides, shouldn't we check the upload was successful?